packages/medusa-payment-stripe/src/core/stripe-base.ts

```diff
@@ -0,0 +1,327 @@
+  ): Promise<PaymentProcessorError | PaymentProcessorSessionResponse> {
+    const intentRequestData = this.getPaymentIntentOptions()
+    const {
+      email,
+      context: cart_context,
+      currency_code,
+      amount,
+      resource_id,
+      customer,
+    } = context
+
+    const description = (cart_context.payment_description ??
+      this.options_?.payment_description) as string
+
+    const intentRequest: Stripe.PaymentIntentCreateParams = {
+      description,
+      amount: Math.round(amount),
+      currency: currency_code,
+      metadata: { resource_id },
+      capture_method: this.options_.capture ? "automatic" : "manual",
+      ...intentRequestData,
+    }
+
+    if (this.options_?.automatic_payment_methods) {
+      intentRequest.automatic_payment_methods = { enabled: true }
+    }
+
+    if (customer?.metadata?.stripe_id) {
+      intentRequest.customer = customer.metadata.stripe_id as string
+    } else {
+      let stripeCustomer
+      try {
+        stripeCustomer = await this.stripe_.customers.create({
+          email,
+        })
+      } catch (e) {
+        return this.buildError(
+          "An error occurred in InitiatePayment during the creation of the stripe customer",
+          e
+        )
+      }
+
+      intentRequest.customer = stripeCustomer.id
+    }
+
+    let session_data
+    try {
+      session_data = (await this.stripe_.paymentIntents.create(
+        intentRequest
+      )) as unknown as Record<string, unknown>
+    } catch (e) {
+      return this.buildError(
+        "An error occurred in InitiatePayment during the creation of the stripe payment intent",
+        e
+      )
+    }
+
+    return {
+      session_data,
+      update_requests: customer?.metadata?.stripe_id
+        ? undefined
+        : {
+            customer_metadata: {
+              stripe_id: intentRequest.customer,
+            },
+          },
+    }
+  }
+
+  async authorizePayment(
+    paymentSessionData: Record<string, unknown>,
+    context: Record<string, unknown>
+  ): Promise<
+    | PaymentProcessorError
+    | {
+        status: PaymentSessionStatus
+        data: PaymentProcessorSessionResponse["session_data"]
+      }
+  > {
+    const status = await this.getPaymentStatus(paymentSessionData)
+    return { data: paymentSessionData, status }
+  }
+
+  async cancelPayment(
+    paymentSessionData: Record<string, unknown>
+  ): Promise<
+    PaymentProcessorError | PaymentProcessorSessionResponse["session_data"]
+  > {
+    try {
+      const id = paymentSessionData.id as string
+      return (await this.stripe_.paymentIntents.cancel(
+        id
+      )) as unknown as PaymentProcessorSessionResponse["session_data"]
+    } catch (error) {
+      if (error.payment_intent?.status === ErrorIntentStatus.CANCELED) {
+        return error.payment_intent
+      }
+
+      return this.buildError(
+        "An error occurred in cancelPayment during the cancellation of the payment",
+        error
```

Comment:
todo: Simplify error message
```suggestion
        "An error occurred in cancelPayment",
        error
```